source/agent/quic/index.js

```diff
@@ -252,36 +238,34 @@ module.exports = function (rpcClient, selfRpcId, parentRpcId, clusterWorkerIP) {
             return callback('callback', {type: 'failed', reason: 'Create Connection failed'});
         }
 
-        connections.addConnection(connectionId, connectionType, options.controller, conn, 'out')
+        router.addLocalDestination(connectionId, connectionType, conn)
         .then(onSuccess(callback), onError(callback));
     };
 
     that.unsubscribe = function (connectionId, callback) {
         log.debug('unsubscribe, connectionId:', connectionId);
-        var conn = connections.getConnection(connectionId);
-        connections.removeConnection(connectionId).then(function(ok) {
-            if (conn && conn.type === 'internal') {
-                internalConnFactory.destroy(connectionId, 'out');
-            } else if (conn) {
-                conn.connection.close();
+        var conn = router.getConnection(connectionId);
+        router.removeConnection(connectionId).then(function(ok) {
+            if (conn) {
+                conn.close();
             }
             callback('callback', 'ok');
         }, onError(callback));
     };
 
-    that.linkup = function (connectionId, audioFrom, videoFrom, dataFrom, callback) {
-        log.debug('linkup.');
-        connections.linkupConnection(connectionId, audioFrom, videoFrom, dataFrom).then(onSuccess(callback), onError(callback));
+    that.linkup = function (connectionId, from, callback) {
+        log.debug('linkup, connectionId:', connectionId, 'from:', from);
+        router.linkup(connectionId, from).then(onSuccess(callback), onError(callback));
     };
 
     that.cutoff = function (connectionId, callback) {
         log.debug('cutoff, connectionId:', connectionId);
-        connections.cutoffConnection(connectionId).then(onSuccess(callback), onError(callback));
+        router.cutoff(connectionId).then(onSuccess(callback), onError(callback));
     };
 
     that.mediaOnOff = function (connectionId, track, direction, action, callback) {
         log.debug('mediaOnOff, connection id:', connectionId, 'track:', track, 'direction:', direction, 'action:', action);
-        var conn = connections.getConnection(connectionId);
+        var conn = router.getConnection(connectionId);
         if (conn) {
             if (conn.type === 'quic') {//NOTE: Only webrtc connection supports media-on-off
                 conn.connection.onTrackControl(track,
```

Comment:
Like previous `conn.connection.close`, the `getConnection` returns the original `conn` added by `addLocalSource`, so use `conn.onTrackControl` instead of `conn.connection.onTrackControl`.
And I think maybe quic agent don't need this `mediaOnOff` RPC?

Reply:
Removed implementation of mediaOnOff. It's not supported by QUIC agent.